core/modules/css/DbInterfaceMySql.cc

```diff
@@ -0,0 +1,126 @@
+// -*- LSST-C++ -*-
+
+/*
+ * LSST Data Management System
+ * Copyright 2019 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "css/DbInterfaceMySql.h"
+
+// System headers
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+#include <stdexcept>
+
+// Third-party headers
+#include "mysql/mysqld_error.h"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+// Qserv headers
+#include "css/CssError.h"
+#include "css/DbInterfaceMySql.h"
+#include "sql/SqlResults.h"
+#include "sql/SqlTransactionStd.h"
+
+using namespace std;
+
+namespace {
+LOG_LOGGER _log = LOG_GET("lsst.qserv.css.DbInterfaceMySql");
+
+} // namespace
+
+namespace lsst {
+namespace qserv {
+namespace css {
+
+
+class KvTransaction : public sql::SqlTransactionStd
+{
+public:
+    // Constructors
+    KvTransaction(sql::SqlConnection& conn)
```

Comment:
This may need to be as well declared as `explicit` to avoid being used in the implicit type conversion. See: https://en.cppreference.com/w/cpp/language/explicit